sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala

```diff
@@ -336,6 +338,86 @@ private[hive] object SparkSQLCLIDriver extends Logging {
       state.close()
     }
   }
+
+  private def getCommandCompleter(context: SQLContext): Array[Completer] = {
+    // StringsCompleter matches against a pre-defined wordlist
+    // We start with an empty wordlist and build it up
+    val candidateStrings = new JArrayList[String]
+    // We add Spark SQL function names
+    // For functions that aren't infix operators, we add an open
+    // parenthesis at the end.
+    context.sessionState.functionRegistry.listFunction().map(_.funcName).foreach { s =>
```

Comment:
shall we only include builtin functions?

Reply:
Do you mean `FunctionRegistry.builtin.listFunction().map(_.funcName)`?

Reply:
yes

Reply:
updated.